kyo-prelude/shared/src/main/scala/kyo2/Coroutine.scala

```diff
@@ -0,0 +1,48 @@
+package kyo2
+
+import kernel.*
+import kyo.Tag
+import scala.collection.immutable.Queue
+
+sealed trait Coroutine[+S] extends Effect[Const[Coroutine.Op[S]], Const[Unit]]
+
+object Coroutine:
+
+    sealed trait Op[-S] derives CanEqual
+    object Op:
+        case object Pause extends Op[Any]
+        abstract class Spawn[-S] extends Op[S]:
+            def apply(): Unit < (Coroutine[S] & S)
+    end Op
+
+    private def erasedTag[S] = Tag[Coroutine[Any]].asInstanceOf[Tag[Coroutine[S]]]
```

Comment:
This is the only part of the effect that isn't fully type-safe. We need it because the kernel still uses `=:=` due to performance concerns with `<:<` to compare tags. Since the `S` is naturally bound to the pending effects of the computation being handled, it's safe to use an erased tag.